src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -61,6 +62,8 @@ internal partial class CircuitHost : IAsyncDisposable
         _circuitHandlers = circuitHandlers ?? throw new ArgumentNullException(nameof(circuitHandlers));
         _logger = logger ?? throw new ArgumentNullException(nameof(logger));
 
+        _dispatchInboundEvent = BuildInboundEventDispatcher(_circuitHandlers);
```

Comment:
We need to respect _circuitHandler.Order here

Reply:
I believe the sorting is done [here](https://github.com/dotnet/aspnetcore/blob/7264c87d64fce4420b382ddf9b834dbc97dd88d2/src/Components/Server/src/Circuits/CircuitFactory.cs#L75), so `_circuitHandlers` is already ordered correctly.